cmake/hipsycl-config.cmake.in

```diff
@@ -73,8 +83,6 @@ elseif(HIPSYCL_PLATFORM MATCHES "cuda|nvidia")
   set(HIPSYCL_PLATFORM_CANONICAL "cuda")
 elseif(HIPSYCL_PLATFORM MATCHES "rocm|amd|hip|hcc")
   set(HIPSYCL_PLATFORM_CANONICAL "rocm")
-else()
-  message(SEND_ERROR "Unknown hipSYCL platform '${HIPSYCL_PLATFORM}'")
```

Comment:
Why remove this?

Reply:
nice catch.

With the new logic, either the user leaves `HIPSYCL_PLATFORM` empty or sets one of the values by cycling down the list of options. So, I thought it is not possible to have any other value. But I now get it that, the environment variable can be a wrong string. I will revert this removal.